dcos/http.py

```diff
@@ -167,9 +173,12 @@ def request(method,
         return response
     elif response.status_code == 401:
         if auth_token is not None:
-            msg = ("Your core.dcos_acs_token is invalid. "
-                   "Please run: `dcos auth login`")
-            raise DCOSAuthenticationException(msg)
+            if prompt_login:
```

Comment:
you have the if/else cases switched. `if prompt_login` we want to call `header_challenge_auth`